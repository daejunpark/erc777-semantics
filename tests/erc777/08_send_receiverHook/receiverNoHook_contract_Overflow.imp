test.preamble();

// send() to other address,  operatorSend() through default operator, operatorSend() from caller.
//  contract holder destination with no hook (account 10), sending overflow
// Recall that account 1 has 10 tokens and 2^256 (a 78 digit number) and is:
// 115792089237316195423570985008687907853269984665640564039457584007913129639936

test.makeCaller(3);
test.makeBalance(115792089237316195423570985008687907853269984665640564039457584007913129639935);
test.send(10, 115792089237316195423570985008687907853269984665640564039457584007913129639926, "");
test.operatorSend(3, 10, 115792089237316195423570985008687907853269984665640564039457584007913129639926, "", ""); //from caller

test.makeCaller(8);
test.operatorSend(3, 10, 115792089237316195423570985008687907853269984665640564039457584007913129639926, "", ""); //through default operator
