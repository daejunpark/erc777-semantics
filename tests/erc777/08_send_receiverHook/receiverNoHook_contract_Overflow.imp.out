Changing the caller to 3
Changing the balance of 3 to 115792089237316195423570985008687907853269984665640564039457584007913129639935
Balance of 3 is 115792089237316195423570985008687907853269984665640564039457584007913129639935
3 is sending 115792089237316195423570985008687907853269984665640564039457584007913129639926 to 10 (data:)
REVERT
Balance of 3 is 115792089237316195423570985008687907853269984665640564039457584007913129639935
Balance of 10 is 100
3 is sending 115792089237316195423570985008687907853269984665640564039457584007913129639926 from 3 to 10 (data:""),  (operatorData:"")
REVERT
Balance of 3 is 115792089237316195423570985008687907853269984665640564039457584007913129639935
Balance of 10 is 100
Changing the caller to 8
8 is sending 115792089237316195423570985008687907853269984665640564039457584007913129639926 from 3 to 10 (data:""),  (operatorData:"")
REVERT
Balance of 3 is 115792089237316195423570985008687907853269984665640564039457584007913129639935
Balance of 10 is 100
